packages/richtext-lexical/src/features/heading/client/index.tsx

```diff
@@ -112,3 +121,22 @@ export const HeadingFeatureClient = createClientFeature<HeadingFeatureProps>(({
     },
   }
 })
+
+const HeadingPlugin: PluginComponent<HeadingFeatureProps> = ({ clientProps }) => {
+  const { enabledHeadingSizes = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'] } = clientProps
+  const lowestAllowed = enabledHeadingSizes.sort().at(-1)
```

Comment:
Just a small nit - we could run `enabledHeadingSizes.sort()` in the feature.server.ts, then the client will receive it already-sorted